modules/library/main/src/main/java/org/geotools/styling/RasterSymbolizerImpl.java

```diff
@@ -64,7 +63,6 @@ public RasterSymbolizerImpl(FilterFactory factory, Description desc, String name
         super(name, desc, (String) null, uom);
         this.filterFactory = factory;
         this.opacity = filterFactory.literal(1.0);
-        this.overlap = filterFactory.literal(OverlapBehavior.RANDOM);
```

Comment:
random is listed as the default in the symbology encoding specification - are you leaving this value as null (as not set).